### PR TITLE
[CL] Add Update Position test

### PR DESCRIPTION
**Pull request by mattverse**
Closes: [#3884](https://github.com/osmosis-labs/osmosis/issues/3884)

## What is the purpose of the change
This PR adds unit tests for the method `UpdatePosition`.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/concentrated-liquidity/lp_test.go
Comment (czarcas7ic) [inline x/concentrated-liquidity/lp_test.go:596]: This is a new position from a new owner. Do you think we should have a test case that calls update position from the same owner that originally created the position?

File: x/concentrated-liquidity/lp_test.go
Comment (czarcas7ic) [inline x/concentrated-liquidity/lp_test.go:623]: should we check what actualAmount0 and actualAmount1 is when we return the error?

File: x/concentrated-liquidity/lp_test.go
Comment (czarcas7ic) [inline x/concentrated-liquidity/lp_test.go:636]: Why do we add liquidityDelta twice to the pools liquidity?

File: x/concentrated-liquidity/lp_test.go
Comment (p0mvn) [inline x/concentrated-liquidity/lp_test.go:567]: We should have more ttest cases such as:
- updating an existing position
- trying to call update position on a non-existent position due to ticks being different
- success case with a positive liquidity delta
- success case with a negative liquidity delta

File: x/concentrated-liquidity/lp_test.go
Comment (mattverse) [reply x/concentrated-liquidity/lp_test.go:636]: this is happening because first time by createPosition, second time by updatePosition: I will leave additional in line comments to make this clearer do you think this makes sense?

File: x/concentrated-liquidity/lp_test.go
Comment (mattverse) [reply x/concentrated-liquidity/lp_test.go:596]: if I understood you correctly, that's what the "happy path" test case does! Would update the test case name to make this clearer if I did understand you correctly

File: x/concentrated-liquidity/lp_test.go
Comment (mattverse) [reply x/concentrated-liquidity/lp_test.go:567]: test case for `updating an existing position` is the first test case if I understood you correctly

Comment (mattverse) [reply x/concentrated-liquidity/lp_test.go:567]: same with positive liquidity delta?

Comment (mattverse) [reply x/concentrated-liquidity/lp_test.go:567]: added other test cases

**Comment by mattverse** [comment]
Thanks for the review @p0mvn @czarcas7ic! Added some more test cases based on the reviews

File: x/concentrated-liquidity/lp_test.go
Comment (p0mvn) [inline x/concentrated-liquidity/lp_test.go:579]: Why does this error?

If we remove liquidity that was added before by making `liquidityDelta` negative, it should pass

Comment (p0mvn) [reply x/concentrated-liquidity/lp_test.go:579]: We should have both error and success cases with negative liquidity but it is still unclear to me why this specific test case errors

File: x/concentrated-liquidity/lp_test.go
Comment (p0mvn) [inline x/concentrated-liquidity/lp_test.go:614]:
```suggestion
			owner:                 s.TestAccs[1], // using a different address makes this a new position
```

**Review by p0mvn — COMMENTED** [review]
LGTM once we figure out the negative liquidity questions.

I think we should also hold off with merging this until Adam's additive tick spacing is merged to avoid breaking that large PR

File: x/concentrated-liquidity/lp_test.go
Comment (mattverse) [reply x/concentrated-liquidity/lp_test.go:579]: oh interesting seems that createPosition is not creating position 🤔 investigating further on

Comment (mattverse) [reply x/concentrated-liquidity/lp_test.go:579]: oh wow the reason that caused this is actually pretty interesting, if we use s.TestAccs[0] in the field and use s.TestAccs[0] directly within the test cases, it would error since the suite setup sets the test accounts to random test accounts everytime

Comment (mattverse) [reply x/concentrated-liquidity/lp_test.go]: Figured this out and added new tests for this

File: x/concentrated-liquidity/lp_test.go
Comment (czarcas7ic) [reply x/concentrated-liquidity/lp_test.go:674]: Exactly, we want to check that they return empty sdk.Ints and not a value

File: x/concentrated-liquidity/lp_test.go
Comment (czarcas7ic) [reply x/concentrated-liquidity/lp_test.go]: Yeah it LGTM now, thank you

File: x/concentrated-liquidity/lp_test.go
Comment (p0mvn) [inline x/concentrated-liquidity/lp_test.go:555]: nit: can you update the spec for `updatePosition` to remove the tests TODO please

File: x/concentrated-liquidity/lp_test.go
Comment (p0mvn) [inline x/concentrated-liquidity/lp_test.go:679]:
```suggestion
```

File: x/concentrated-liquidity/lp_test.go
Comment (p0mvn) [inline x/concentrated-liquidity/lp_test.go:684]: Let's also validate tick updates. There is a helper for that called `validateTickUpdates`

**Review by p0mvn — APPROVED** [review]
LGTM once comments are addressed. Nice work

File: x/concentrated-liquidity/lp_test.go
Comment (mattverse) [reply x/concentrated-liquidity/lp_test.go:555]: done.

File: x/concentrated-liquidity/lp_test.go
Comment (mattverse) [reply x/concentrated-liquidity/lp_test.go:686]: done

File: x/concentrated-liquidity/lp_test.go
Comment (mattverse) [reply x/concentrated-liquidity/lp_test.go:679]: gotcha, added

**Pull request merged**
